standards.md

```diff
@@ -1,47 +1,126 @@
+  self-contained and makes sense without the context of the others in the PR
 
-- Start with a subject line
-- Contain a body that explains _why_ you're making the change you're making
-- Reference an issue number one exists, closing it if applicable (with text such
-  as
-  ["Fixes #245" or "Closes #111"](https://help.github.com/articles/closing-issues-using-keywords/))
+## Docs
 
-Aim for [2 paragraphs in the body](https://www.youtube.com/watch?v=PJjmw9TRB7s).
-Not sure what to put? Include:
+* Include Markdown doc updates for user visible features
+* Spelling and grammar should be correct
+* Try to make formatting look as good as possible (use preview mode to check)
+* Follow [content](https://github.com/tektoncd/website/blob/master/content/en/doc-con-content.md)
+  and [formatting](https://github.com/tektoncd/website/blob/master/content/en/doc-con-formatting.md) guidelines
+* Should explain thoroughly how the new feature works
 
-- What is the problem being solved?
-- Why is this the best approach?
-- What other approaches did you consider?
-- What side effects will this approach have?
-- What future work remains to be done?
+## Tests
 
-## Coding standards
+* New features (and often/whenever possible bug fixes) have one or both of:
+  * Examples (i.e. yaml tests)
+  * End to end tests
+  * Reconciler tests
+* New types have:
+  * Validation + validation tests
+* Unit tests:
+  * Coverage should remain the same or increase
+  * Test exported functions only, in isolation:
+    * Each exported function should have tests
+    * Each test should only test one function at a time
+    * If you find yourself wanting to test an unexported function, consider whether
+      it would make sense to move the test into another package and export it
+* Test code
+  * When using cmp.Diff the argument order is always (want, got) and in the error message include (-want +got)
+    (and/or use a lib like [PrintWantGot](https://github.com/tektoncd/pipeline/blob/master/test/diff/print.go))
+  * Table driven tests: do not use the same test for success and fail cases if the logic is different
+    (e.g. do not have two sets of test logic, gated with `wantErr`)
 
-### Go
+## API Design
+
+* Avoid kubernetes specific feature in our APIs as much as possible
+* If we don’t definitely need the feature, don’t add it
+  * Must be backed up with [CI/CD specific use cases](https://github.com/tektoncd/community/blob/master/roadmap.md#mission-and-vision)
+* Prefer keeping the API simple:
+  * If you can avoid adding a new feature, don’t add it
+  * Prefer reusing existing features to adding new ones
+
+## Functionality
+
+* It should be safe to cut a release at any time, i.e. merging this PR should not
+  put us into an unreleasable state
+    * When incrementally adding new features, this may mean that a release could contain
+      a partial feature, i.e. the type specification only but no functionality
```

Comment:
It may be worth adding something about how we document this.
If a feature includes a new API or CRD we might want to document that it is still not functional?